app/src/main/java/ai/elimu/vitabu/ui/storybook/StoryBookActivity.kt

```diff
@@ -47,6 +53,42 @@ class StoryBookActivity : AppCompatActivity() {
 
         val zoomOutPageTransformer = ZoomOutPageTransformer()
         binding.viewPager.setPageTransformer(true, zoomOutPageTransformer)
+        binding.viewPager.setOnPageChangeListener(object : ViewPager.OnPageChangeListener {
+
+            override fun onPageScrolled(
+                position: Int,
+                positionOffset: Float,
+                positionOffsetPixels: Int
+            ) {
+            }
+
+            override fun onPageSelected(position: Int) {
+                if (position == storyBookChapters.size - 1 && !shouldReportCompletion) {
+                    completedStoryBook = ContentProviderUtil.getStoryBookGson(
+                        storyBookId,
+                        applicationContext, BuildConfig.CONTENT_PROVIDER_APPLICATION_ID
+                    ) ?: return
+
+                    shouldReportCompletion = true
+                }
+            }
+
+            override fun onPageScrollStateChanged(state: Int) {
+            }
+
+        })
+    }
+
+    override fun onStop() {
```

Comment:
@tuancoltech Can we change from `onStop()` to [`onDestroy()`](https://developer.android.com/reference/android/app/Activity#onDestroy())? (Since an activity can be restarted after `onStop`).